### PR TITLE
feat: subcommands type-check only local files by default

### PR DESCRIPTION
**Pull request by bartlomieju**
Closes https://github.com/denoland/deno/issues/14498

### CODE REVIEW

File: cli/flags.rs [hunk 1]
```diff
@@ -2041,10 +2046,13 @@ fn no_check_arg<'a>() -> Arg<'a> {
...
-    .help("Skip type checking modules")
+    .help("DEPRECATED: Skip type checking modules")
     .long_help(
-      "Skip type-checking. If the value of '--no-check=remote' is supplied, \
-      diagnostic errors from remote modules will be ignored.",
+      "DEPRECATED: Skip type-checking. If the value of '--no-check=remote' is supplied,
+diagnostic errors from remote modules will be ignored.
+
+This option will be removed in Deno 1.23, see `--check` instead.
+      ",
```
Comment (ry) [inline cli/flags.rs:2055]: this seems to be doing more than just "type-check only local files by default"

**Review by ry — COMMENTED** [review]
I would expect a test that shows, for example, that `deno test` can have type errors in remote modules that do not cause error.

File: cli/flags.rs (hunk 1 above)
Comment (bartlomieju) [reply cli/flags.rs:2055]: I don't understand what you mean here

**Comment by bartlomieju** [comment]
@ry do you want tests for all subcommands?

**Comment by ry** [comment]
@bartlomieju which subcommands are effected? `deno test`... ? Yes I think we should probably have tests for each.

**Comment by bartlomieju** [comment]
Following commands are affected by this PR:
- `deno bench`
- `deno bundle`
- `deno cache`
- `deno compile`
- `deno eval`
- `deno install`
- `deno run`
- `deno test`

**Review by ry — APPROVED** [review]
LGTM

**Review by dsherret — APPROVED** [review]
LGTM

**Review by littledivy — APPROVED** [review]

**Pull request merged**